### PR TITLE
Enable setting of ND properties in CrystalMapProperties

### PR DESCRIPTION
**Pull request by hakonanes**
Signed-off-by: Håkon Wiik Ånes <hwaanes@gmail.com>

Todo:
- [x] Ensure these ND arrays are written to the orix HDF5 file correctly (add tests for this)

With this small change in `CrystalMapProperties` we can now add properties with more than one value in each map point, enabling e.g. to store multiple best matching indices from a list of simulated patterns.

Tests are added.

(Note that even if a map is 2D [which it usually is], properties are always stored as 1D arrays.)

```python
>>> import numpy as np
>>> from orix.crystal_map.crystal_map_properties import CrystalMapProperties
>>> map_size = 10
>>> d = {"prop_1d": np.arange(map_size)}  # 1D
>>> props = CrystalMapProperties(d, id=np.arange(map_size))
>>> props
{'prop_1d': array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])}
>>> props["prop_2d"] np.arange(map_size * 2).reshape((map_size, 2))
>>> props
{'prop_1d': array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]),
 'prop_2d': array([[ 0,  1],
        [ 2,  3],
        [ 4,  5],
        [ 6,  7],
        [ 8,  9],
        [10, 11],
        [12, 13],
        [14, 15],
        [16, 17],
        [18, 19]])}
>>> props["prop_2d_wrong_shape"] = np.arange(map_size * 2).reshape((2, map_size))
Traceback (most recent call last):
  File "/home/hakon/miniconda3/envs/orix-dev/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3417, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)
  File "<ipython-input-10-ec06fc2f2380>", line 1, in <module>
    props["prop_2d_wrong"] = np.arange(map_size * 2).reshape((2, map_size))
  File "/home/hakon/kode/orix/orix/crystal_map/crystal_map_properties.py", line 81, in __setitem__
    array[self.is_in_data, ...] = value
IndexError: boolean index did not match indexed array along dimension 0; dimension is 2 but corresponding boolean dimension is 10
```

The first axis of the new array must have the same length as the map size (I've added this explanation to the `CrystalMapProperties.__setitem__()` docstring).

@onatlandsmyr: could you pull this branch of orix into your kikuchipy dev environment and see that it works as expected? After you've done that, I'll remove the WIP label and this can be reviewed and merged.

Although I initially set CrystalMapProperties up to only allow 1D arrays, I consider this a bug since it obviously should allow for ND arrays. I hope this can be part of a patch release, since we need this in kikuchipy for v0.3.

### CODE REVIEW

**Comment by onatlandsmyr** [comment]
The fix works as expected for my use cases.

**Comment by hakonanes** [comment]
Travis says the PR build is queued... Have we used up our quota for the day?

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,4 +9,5 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - This project now keeps a Changelog
 
 ### Fixed
+- CrystalMap properties allows > 2D arrays
```
Comment (pc494) [inline CHANGELOG.md:12]: This is ambiguous, I assume it's `CrystalMap properties allow arrays with dimensions greater than 2`? I've got too used to using `>` as an arrow...

**Review by pc494 — COMMENTED** [review]
A single CHANGELOG question

File: CHANGELOG.md (hunk 1 above)
Comment (hakonanes) [reply CHANGELOG.md]: Yup, changed to your improved wording.

**Review by pc494 — APPROVED** [review]
Will merge once the tests pass.

**Comment by hakonanes** [comment]
Thanks a lot for the quick reviews and merges of these bug fixes!

**Pull request merged**